### PR TITLE
Add type hints in vgg.py

### PR DESCRIPTION
**Pull request by nxznm**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add type annotation for mmcv mentioned in #1948.

## Modification

Add type hints in `mmcv/cnn/vgg.py`.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/cnn/vgg.py [hunk 1]
```diff
@@ -138,7 +140,8 @@ def init_weights(self, pretrained=None):
         else:
             raise TypeError('pretrained must be a str or None')
 
-    def forward(self, x):
+    def forward(self,
+                x: torch.Tensor) -> Union[torch.Tensor, Tuple[torch.Tensor]]:
```
Comment (zhouzaida) [inline mmcv/cnn/vgg.py:144]:
```suggestion
    def forward(self, x: Tensor) -> Union[Tensor, Tuple[Tensor]]:
```

**Comment by zhouzaida** [comment]
Hi @nxznm , thanks for your contribution. After resolving the above comments, this PR can be merged.

File: mmcv/cnn/vgg.py [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+from typing import Optional, Sequence, Tuple, Union
 
 import torch.nn as nn
+from torch import Tensor
 
 from .utils import constant_init, kaiming_init, normal_init
 
 
-def conv3x3(in_planes, out_planes, dilation=1):
+def conv3x3(in_planes: int, out_planes: int, dilation: int = 1):
```
Comment (HAOCHENYE) [inline mmcv/cnn/vgg.py:11]: Type hints for return value

File: mmcv/cnn/vgg.py [hunk 3]
```diff
@@ -138,7 +140,7 @@ def init_weights(self, pretrained=None):
         else:
             raise TypeError('pretrained must be a str or None')
 
-    def forward(self, x):
+    def forward(self, x: Tensor) -> Union[Tensor, Tuple[Tensor]]:
```
Comment (zhouzaida) [inline mmcv/cnn/vgg.py:143]:
```suggestion
    def forward(self, x: Tensor) -> Union[Tensor, Tuple[Tensor, ...]]:
```

**Review by zhouzaida — APPROVED** [review]

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @nxznm ！First of all, we want to express our gratitude for your significant PR in the mmcv project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/raweFPmdzG

If you have WeChat，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**